src/CoreWCF.Queue/src/CoreWCF/Queue/Common/QueueHandshakeMiddleware.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using CoreWCF.Queue.Common.Configuration;
+
+namespace CoreWCF.Queue.Common
+{
+    public class QueueHandShakeMiddleWare
+    {
```

Comment:
I don't like the use of the word HandShake here. This comes from NetTcp, but that has an actual connection handshake.Maybe just call is QueueMiddleware. I think this can be internal for now.